torch/fx/experimental/symbolic_shapes.py

```diff
@@ -3665,6 +3674,43 @@ def _maybe_guard_eq(self, expr: Union["sympy.Eq", "sympy.Ne"], concrete_bool: bo
         free = sorted(free, key=lambda x: (self.size_hint(x, allow_none=True) or sys.maxsize, x.name), reverse=True)  # type: ignore[attr-defined]
         lhs = expr.lhs
         rhs = expr.rhs
+
+        if not is_eq:
```

Comment:
Is the equality case handled elsewhere? Do we just merge value ranges with `&` for them? Or is that unsafe?

Reply:
The equality stuff is the pre-existing code in this function, below. It indirectly updates value ranges because when we are able to solve the equality into a replacement, the variable is eliminated and so you get updated value ranges.

Your comment here makes me realize that in principle, we could do better on multi-variate expressions by trying to progressively solve for each variable and propagate ranges for each one we are able to solve, but IDK, seems cursed.

Reply:
That would propagate ranges nicely I think. One should do it until it finds a fixed point, but that should only take 1-2 iters at most.